Controllers/Backend/ViewSnapshots.php

```diff
@@ -55,4 +58,68 @@ public function listAction()
             ['success' => true, 'data' => $data, 'total' => $total]
         );
     }
+
+    /**
+     * @throws ParameterMissingException
+     * @throws Exception
+     */
+    public function diffAction()
+    {
+        $sessionFrom = $this->Request()->getParam('sessionFrom');
+        $stepFrom = $this->Request()->getParam('stepFrom');
+        $sessionTo = $this->Request()->getParam('sessionTo', $sessionFrom);
+        $stepTo = $this->Request()->getParam('stepTo');
+
+        if (empty($sessionFrom)) {
+            throw new ParameterMissingException('sessionFrom is missing');
+        }
+
+        if (empty($stepFrom)) {
+            throw new ParameterMissingException('stepFrom is missing');
+        }
+
+        if (empty($sessionTo)) {
+            throw new ParameterMissingException('sessionTo is missing');
+        }
+
+        if (empty($stepTo)) {
+            throw new ParameterMissingException('stepTo is missing');
+        }
+
+        /** @var Diff $differ */
+        $differ = $this->get('frosh_view_snapshots.services.diff');
+        $dataFrom = $this->getSnapshotStep($sessionFrom, $stepFrom);
+        $dataTo = $this->getSnapshotStep($sessionTo, $stepTo);
+
+        $this->View()->assign([
+            'success' => true,
+            'data' => [
+                'sessionID' => $differ->diffPlain($dataFrom['sessionID'], $dataTo['sessionID'])->renderDiffToHTML(),
+                'template' => $differ->diffPlain($dataFrom['template'], $dataTo['template'])->renderDiffToHTML(),
+                'step' => $differ->diffPlain($dataFrom['step'], $dataTo['step'])->renderDiffToHTML(),
+                'variables' => $differ->diffSerialized($dataFrom['variables'], $dataTo['variables'])->renderDiffToHTML(),
+                'params' => $differ->diffJson($dataFrom['params'], $dataTo['params'])->renderDiffToHTML(),
+            ],
+        ]);
+    }
+
+    /**
+     * @throws Exception
```

Comment:
Please add your parameters to the PHPDoc

Reply:
As of PHP 7 (as stated in README.md) I do not have to add PHPDoc for auto completion. Why do I need redundant type hints?

Reply:
We're open for discussion here. We always put the parameters in the PHPDoc as well in our company but I understand your argument, too. Some other opinions here? @shyim @dneustadt @kleinmann @jochenmanz ?

Reply:
We're  only adding type hints as PHPDoc if they add more value, e.g. `int[]` instead of only `array` in the signature. If the information is redundant, I support not having it in the PHPDoc.

Reply:
Regarding the [comment](https://github.com/FriendsOfShopware/FroshViewSnapshots/pull/5#discussion_r215887677) from @dneustadt I will use PHP5.6 valid signatures as the plugin shall keep backward compatibility. Therefore I will add some more PHPDoc.